### PR TITLE
Allow empty slot-scope to render a scoped slot instead of a normal one

### PR DESCRIPTION
**Pull request by basselAhmed**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch for v2.x (or to a previous version branch), _not_ the `master` branch
- [x] All tests are passing: https://github.com/vuejs/vue/blob/dev/.github/CONTRIBUTING.md#development-setup
- [x] New/updated tests are included

If adding a **new feature**, the PR's description includes:
- [x] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

Please check issue #8912 and #8496 
This PR allows rendering a single slot multiple times by passing an empty slot-scope which makes the slot be rendered as a scoped slot instead of a normal one.

### CODE REVIEW

File: src/compiler/parser/index.js [hunk 1]
```diff
@@ -476,7 +476,10 @@ function processSlot (el) {
         )
       }
       el.slotScope = slotScope || getAndRemoveAttr(el, 'slot-scope')
-    } else if ((slotScope = getAndRemoveAttr(el, 'slot-scope'))) {
+    } else if (
+        (slotScope = getAndRemoveAttr(el, 'slot-scope')) ||
+        (slotScope = getAndRemoveAttr(el, 'slot-scope')) === ''
```
Comment (posva) [inline src/compiler/parser/index.js]: shouldn't we be able to do `slotScope === ''` as the second check?

Comment (posva) [reply src/compiler/parser/index.js]: I wonder if we could only check against `!= null`

Comment (basselAhmed) [reply src/compiler/parser/index.js]: Yes, we could do that as the second check, I don't think null is the only falsy value we should check against @posva

Comment (posva) [reply src/compiler/parser/index.js]: it's `null` and `undefined`, but yeah, I didn't check further, just dropped the idea haha

**Comment by KaelWD** [comment]
Vue 2.6 now has `v-slot` instead, this can be closed.

**Pull request closed without merging**